src/pyconcrete/__init__.py

```diff
@@ -77,47 +54,39 @@ def load_module(self, fullname):
             # reference http://stackoverflow.com/questions/1830727/how-to-load-compiled-python-modules-from-memory
             # MAGIC + TIMESTAMP
             magic = 8
-
-        code = marshal.loads(data[magic:])
-
-        m = self.new_module(fullname, self.full_path, self.pkg_path)
-        sys.modules[fullname] = m
-        exec(code, m.__dict__)
-        return m
-
-    def is_package(self, fullname):
-        return self.is_pkg
+        return magic
 
     @staticmethod
     def _validate_version(data):
         magic = imp.get_magic()
         ml = len(magic)
         if data[:ml] != magic:
-            import struct
-
             # convert little-endian byte string to unsigned short
             py_magic = struct.unpack('<H', magic[:2])[0]
             pye_magic = struct.unpack('<H', data[:2])[0]
             raise ValueError("Python version doesn't match with magic: python(%d) != pye(%d)" % (py_magic, pye_magic))
 
+    def get_code(self, fullname):
+        if not self.path.endswith(EXT_PYE):
+            return super().get_code(fullname)
+
+        path = self.get_filename(fullname)
+        data = decrypt_buffer(self.get_data(path))
+        self._validate_version(data)
+        return marshal.loads(data[self.magic:])
+
+    def get_source(self, fullname):
+        if self.path.endswith(EXT_PYE):
+            return None
+        return super().get_source(fullname)
+
 
-class PyeMetaPathFinder(object):
-    def find_module(self, fullname, path=None):
-        mod_name = fullname.split('.')[-1]
-        paths = path if path else sys.path
+loader_details = [(PyeLoader, SOURCE_SUFFIXES)] + _get_supported_file_loaders()
```

Comment:
Does it is a good approach to use the python internal function `_get_supported_file_loaders()`? Does there is any other alternatives?

Reply:
I agree that internal functions are evil, but there is not much choice here, we have to extend and "override" the standard path hook, which is installed using the `_install()` internal function of the `_bootstrap_external` module. So we can either use internal functions or copy/monkey-patch the code of the internal function, which is also bad.

Reply:
Agree with your perspective. I also checked python3.8~3.13 the function `_bootstrap_external._get_supported_file_loaders()` all exists.